### PR TITLE
[PDI-20085] - KETTLE_EMPTY_STRING_DIFFERS_FROM_NULL produces opposite display for NULLS

### PR DESCRIPTION
**Pull request by andreramos89**
@smmribeiro 
@renato-s 
@befc

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 15m 38s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **1270**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9311/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by befc — APPROVED** [review]
Overall the change seems fine, although I'd need further context to understand the full ramifications of the order of the conditions in the `if` statements. But the unit test added seems to make the right validation for the bug fix intended. 👌 

Added just some minor comments pertaining to code maintainability

File: core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 1]
```diff
@@ -4044,9 +4045,9 @@ public Object convertDataFromString( String pol, ValueMetaInterface convertMeta,
     Boolean normalizeNullStringToEmpty = !convertStringToBoolean(
       Const.NVL( System.getProperty( Const.KETTLE_DO_NOT_NORMALIZE_NULL_STRING_TO_EMPTY, "N" ), "N" ) );
```
[the remark this answers is not in the document]
Comment (andreramos89) [reply core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java]: Done.

File: core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 2]
```diff
@@ -4044,9 +4045,9 @@ public Object convertDataFromString( String pol, ValueMetaInterface convertMeta,
...
 
-    if ( normalizeNullStringToEmpty ) {
-      if ( pol == null && isStringValue && isEmptyAndNullDiffer ) {
-        pol = Const.NULL_STRING;
+    if ( isEmptyAndNullDiffer && pol == null && isStringValue ) {
+      if( normalizeNullStringToEmpty ) {
+        pol = StringUtils.EMPTY;
```
[the remark this answers is not in the document]
Comment (andreramos89) [reply core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java]: added comment. About the logic of the if statement, the idea is to replicate what is explained in the kettle properties file description for the properties.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 30s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **1270**, Failures: **0**, Skipped: **1** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9311/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**